### PR TITLE
OCLOMRS-406: implement edit mapping on the right interface

### PR DESCRIPTION
**Pull request by GodswillOnuoha**
# JIRA TICKET NAME:
[implement edit mapping on the right interface](https://issues.openmrs.org/browse/OCLOMRS-406)

# Summary:
"edit mapping" is currently implemented in the view mappings interface. It needs to be implemented in the edit concept interface

### CODE REVIEW

File: src/tests/dictionaryConcepts/container/EditConcept.test.jsx [hunk 1]
```diff
@@ -189,3 +190,121 @@ describe('Test suite for dictionary concepts components', () => {
...
+
+  it('should call removeMappingRow function', () => {
+    const event = { target: 0 };
+    const instance = wrapper.find('EditConcept').instance();
+    instance.removeMappingRow(event);
+  });
+
+  it('should call updateAsyncSelectValue function', () => {
+    const value = { index: 0, value: 'malaria 1', label: 'malaria 1' };
+    const instance = wrapper.find('EditConcept').instance();
```
Comment (dkayiwa) [inline src/tests/dictionaryConcepts/container/EditConcept.test.jsx]: Shouldn't we expect something?

Comment (GodswillOnuoha) [reply src/tests/dictionaryConcepts/container/EditConcept.test.jsx]: updated

**Comment by dkayiwa** [comment]
Did you look into the coverage decrease?

File: src/tests/dictionaryConcepts/components/CreateMapping.test.js [hunk 2]
```diff
@@ -45,4 +49,31 @@ describe('Test suite for dictionary concepts components', () => {
...
+    wrapper = mount(<Router>
+      <table><tbody><CreateMapping {...newProps} /></tbody></table>
+    </Router>);
+    const inputs = wrapper.find('input');
+    expect(inputs).toHaveLength(3);
+  });
+
+  it('should render when isNew is true and source equal to CIEL', () => {
+    const newProps = {
+      ...props,
```
Comment (dkayiwa) [inline src/tests/dictionaryConcepts/components/CreateMapping.test.js]: Where are you passing the source?

File: src/tests/dictionaryConcepts/container/EditConcept.test.jsx [hunk 3]
```diff
@@ -189,3 +190,127 @@ describe('Test suite for dictionary concepts components', () => {
...
+    <EditConcept {...props} />
+  </Router>);
+
+  it('should call addMappingRow function', () => {
+    const instance = wrapper.find('EditConcept').instance();
+    instance.addMappingRow();
+  });
+
+  it('should call removeMappingRow function', () => {
+    const event = { target: 0 };
```
Comment (dkayiwa) [inline src/tests/dictionaryConcepts/container/EditConcept.test.jsx]: Shouldn't you expect something?

Comment (GodswillOnuoha) [reply src/tests/dictionaryConcepts/container/EditConcept.test.jsx]: updated

File: src/tests/dictionaryConcepts/container/EditConcept.test.jsx [hunk 4]
```diff
@@ -189,3 +190,127 @@ describe('Test suite for dictionary concepts components', () => {
...
+      }],
+    },
+    newRow: '1234',
+  };
+  const wrapper = mount(<Router>
+    <EditConcept {...props} />
+  </Router>);
+
+  it('should call addMappingRow function', () => {
+    const instance = wrapper.find('EditConcept').instance();
```
[the remark this answers is not in the document]
Comment (GodswillOnuoha) [reply src/tests/dictionaryConcepts/container/EditConcept.test.jsx]: updated

[the remark this answers is not in the document]
Comment (GodswillOnuoha) [reply src/tests/dictionaryConcepts/container/EditConcept.test.jsx]: updated

File: src/tests/dictionaryConcepts/components/CreateMapping.test.js (hunk 2 above)
Comment (GodswillOnuoha) [reply src/tests/dictionaryConcepts/components/CreateMapping.test.js]: `source` is defined in the `props`, at the beginning of this file

File: src/tests/dictionaryConcepts/components/CreateMapping.test.js [hunk 5]
```diff
@@ -45,4 +49,31 @@ describe('Test suite for dictionary concepts components', () => {
...
+      isNew: true,
+      source: 'Snomed',
+    };
+
+    wrapper = mount(<Router>
+      <table><tbody><CreateMapping {...newProps} /></tbody></table>
+    </Router>);
+    const inputs = wrapper.find('input');
+    expect(inputs).toHaveLength(3);
```
Comment (dkayiwa) [inline src/tests/dictionaryConcepts/components/CreateMapping.test.js]: Would this fail if i changed the above source to CIEL?

Comment (dkayiwa) [reply src/tests/dictionaryConcepts/components/CreateMapping.test.js]: Would this fail if i changed the above isNew to false?

Comment (GodswillOnuoha) [reply src/tests/dictionaryConcepts/components/CreateMapping.test.js]: Yes. Changing the source to CIEL or isNew to false causes the assertion to fail

Comment (dkayiwa) [reply src/tests/dictionaryConcepts/components/CreateMapping.test.js]: Would the test fail if i changed the source to ciel?

Comment (GodswillOnuoha) [reply src/tests/dictionaryConcepts/components/CreateMapping.test.js]: It doesn't fail if source is changed to ciel

Comment (dkayiwa) [reply src/tests/dictionaryConcepts/components/CreateMapping.test.js]: Does that mean these names are case sensitive?

Comment (GodswillOnuoha) [reply src/tests/dictionaryConcepts/components/CreateMapping.test.js]: there current implementation is case sensitive, but an issue has been raised to fix that

Comment (GodswillOnuoha) [reply src/tests/dictionaryConcepts/components/CreateMapping.test.js]: I can't login to JIRA to confirm that. But I'm sure Darius pointed that out. I will create the ticket if it doesn't exist.

**Pull request merged**